### DISCUSSION TITLE
Insert queries should return the Entity, aren't they?

### DISCUSSION THREAD

**Discussion by Disregard-Therest** [serverpod/serverpod#1710] [Q&A] [answered]
Hi, guys,
for me it's pretty straightforward that after inserting an entity, I need to return it to front in order to work with further.
At least in order to have it id for referencing.

Currently, I need to perform additional request in order to get the id.

Is it ok?

**Comment by Frank1016** [comment, marked as answer]
Yes, you are right, for current version 1.1.1, you need another request for the id or the whole entity. However, for Serverpod's next version 1.2, this will be changed, and the entity object will be returned directly after inserting operation:

https://docs.serverpod.dev/next/concepts/database/crud

![image](https://github.com/serverpod/serverpod/assets/32797520/272c1cb6-6a82-4a58-9879-f5243c37f61d)

**Reply by Isakdl** [reply]
Actually, in version 1.1.1 the input object is mutated and will have the `id` field set after the method is called.

In version 1.2+ the entire object is returned and the input is NOT mutated.

**Reply by Frank1016** [reply]
Hi @Isakdl , when can we expect the 1.2 version of Serverpod to be released?

**Reply by Isakdl** [reply]
https://www.youtube.com/watch?v=AmHZP6T1NdQ ;)

**Reply by Frank1016** [reply]
In 7 days!!! Fantastic! Can't wait to try the new features! Thank you!
